src/fhirtypes/StructureDefinition.ts

```diff
@@ -316,6 +437,21 @@ export class StructureDefinition {
     return pathParts;
   }
 
+  /**
+   * Tests to see if the last bracket in a PathPart is a positive int, and if so returns it
+   * @param {PathPart} pathPart - The part of the path to test
+   * @returns {number} The index if it exists and is positive, otherwise undefined
+   *
+   */
+  private getArrayIndex(pathPart: PathPart): number {
+    const lastBracket = pathPart.brackets?.slice(-1)[0];
+    let arrayIndex: number;
+    if (/[0]|[-+]?[1-9][0-9]*/.test(lastBracket)) {
```

Comment:
I think this regex may be more confusing than it needs to be...  and isn't quite correct, in that it would match on anything with a number in it (for example, it would match on `'abc25xyz'`).  I don't see why we shouldn't allow leading zeros (e.g., `01`), even if they are a little weird.  So fixing it up to not allow those other non-digit characters, and to allow leading zeroes (mainly to simplify the regex), we would have something like:
```
if (/^[-+]?\d+$/.test(lastBracket)) {
```